augurpy/calculate_AUC.py

```diff
@@ -71,39 +75,110 @@ def draw_subsample(input: DataFrame, augur_mode: str, subsample_size: int, strat
             permuting the labels
         subsample_size: number of cells to subsample randomly per type from each experimental condition
         stratified: if `True` subsamples are stratified according to condition
+        random_state: set random seed
 
     Returns:
         Subsample of input of size subsample_size
     """
-    pass
+    if augur_mode == "permut":
+        # shuffle labels
+        y_columns = [col for col in adata.obs if col.startswith("y")]
+        adata.obs[y_columns] = adata.obs[y_columns].sample(frac=1, random_state=random_state).reset_index(drop=True)
 
+    if augur_mode == "velocity":
+        # no feature selection, assuming this has already happenend in calculating velocity
+        subsample = sc.pp.subsample(adata, n_obs=subsample_size, copy=True, random_state=random_state)
+        return subsample
 
-def subsample_cross_validate(
-    input: DataFrame,
-    augur_mode: str,
-    subsample_size: int,
-    stratified: bool,
-    estimator: Union[RandomForestRegressor, RandomForestClassifier, LogisticRegression],
-    metrics: list,
-    folds: int,
-    subsample_idx: int,
-) -> DataFrame:
-    """Cross validate subsample.
+    # randomly sample features
+    # random.seed()
```

Comment:
Should be affected by the passed seed, no? Also, maybe wrap the last block into an `else` block?